src/mailroom.py

```diff
@@ -0,0 +1,196 @@
+    """Main func that get executed when run in the CLI."""
+    try:
+        while True:
+            clear_screen()
+            print(welcome_message())
+            user_input = input("Selection: ")
+            if user_input not in ['1', '2', '3']:
+                continue
+            handle_input(user_input)()
+    except KeyboardInterrupt:
+        exit()
+
+
+def send_thanks():  # pragma: no cover
+    """Send thank-you note submenu."""
+    while True:
+        donor = input("Enter Donor name, 'list' or 'return': ")
+        if donor == 'list':
+            clear_screen()
+            print(build_report_table(donor_list_by_total(donor_dict)))
+        elif donor == 'return':
+            clear_screen()
+            return
+        elif donor == '':
+            continue
+        else:
+            amount = 0
+            while not amount:
+                amount = validate_donation(
+                    input("Input donation amount: "))
+            else:
+                donor_dict.setdefault(donor, []).append(amount)
+            print('\n' + generate_thankyou(donor, format_amount(amount)))
+
+
+def report_donors():    # pragma: no cover
+    """Report submenu."""
+    clear_screen()
+    donor_list = donor_list_by_total(donor_dict)
+    print(build_report_table(donor_list))
+
+
+def report_donors_wait():    # pragma: no cover
+    """Report submenu."""
+    report_donors()
+    press_to_continue()
+
+
+def exit():     # pragma: no cover
+    """Terminate program without error."""
+    print(save_donors(donor_dict))
+    print('Script terminated.\n')
+    sys.exit(0)
+
+
+def clear_screen():     # pragma: no cover
+    """Clear screen."""
+    os.system('cls' if os.name == 'nt' else 'clear')
+
+
+def press_to_continue():        # pragma: no cover
+    """Wait for user input before proceeding."""
+    input('Press any button to continue.')
+
+
+def welcome_message():
+    """Return welcome message and instruction when the app starts."""
+    return ('Welcome to the donor database.\n'
+            'Enter 1 to send thank you note\n'
+            'Enter 2 to view report\n'
+            'Enter 3 to exit\n')
+
+
+def read_donors(donor_file_name='donor.txt'):
+    """Read donor data from a txt file."""
+    donor_dict = {}
+    try:
+        donor_file = io.open('{}/{}'.format(
+            os.path.dirname(os.path.abspath(__file__)),
+            donor_file_name
+        ), 'r')
+        for line in donor_file:
+            temp_list = line.split('||')
+            donor = temp_list.pop().rstrip()
+            donor_dict[donor] = [float(i) for i in temp_list]
+        donor_file.close()
+    except IOError:
+        donor_dict = DEFAULT_DONORS
+    return donor_dict
+
+
+def save_donors(donor_dict, donor_file_name='donor.txt'):
+    """Save donors information to a test file."""
+    try:
+        donor_file = open('{}/{}'.format(
+            os.path.dirname(os.path.abspath(__file__)), donor_file_name
+        ), 'w')
```

Comment:
nice!  You'll be ahead of the game at the end of this week when it comes to serving files with HTTP!